packages/repository/src/define-repository-class.ts

```diff
@@ -3,28 +3,28 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import {Constructor} from '@loopback/context';
 import assert from 'assert';
+import {PrototypeOf} from './common-types';
 import {Entity, Model} from './model';
 import {
-  CrudRepository,
   DefaultCrudRepository,
   DefaultKeyValueRepository,
-  EntityCrudRepository,
   juggler,
-  KeyValueRepository,
   Repository,
 } from './repositories';
 
 /**
- * Signature for CrudRepository classes
+ * Signature for a Repository class bound to a given model. The constructor
+ * accepts only the dataSource to use for persistence.
+ *
+ * `define*` functions return a class implementing this interface.
  *
  * @typeParam M - Model class
  * @typeParam R - Repository class/interface
  */
-export interface CrudRepositoryClass<
+export interface NamedRepositoryClass<
```

Comment:
There is no need to have specialized types for KeyValue and CRUD repositories, we can use a generic Repository type as the starting point.

I am not comfortable using `RepositoryClass` name, because it's too ambiguous and possibly confusing, since it uses different constructor argument than `PersistedModelClass` and friends.

I was looking for a name that would describe the fact that the repository class is already bound to a given model. `ModelRepositoryClass` did not feel right either, so I settled on `NamedRepositoryClass`. Feel free to propose a better name.

Reply:
If it's `NamedRepositoryClass`, should we add `name: string` explicitly?

What about `GeneratedRepositoryClass`?

Reply:
Maybe we should take another angle. My intention is to distinguish between a generic repository class that works for any Model/Entity and a repository class that's already specialized for (bound to) a given Model/Entity.

In that light, perhaps `BoundRepositoryClass` or `SpecializedRepositoryClass` would be better names?